Source/Core/Common/TraversalClient.cpp

```diff
@@ -0,0 +1,374 @@
+// This file is public domain, in case it's useful to anyone. -comex
+
+#include "TraversalClient.h"
+#include "enet/enet.h"
+#include "Timer.h"
+
+static void GetRandomishBytes(u8* buf, size_t size)
+{
+	// We don't need high quality random numbers (which might not be available),
+	// just non-repeating numbers!
+	srand(enet_time_get());
+	for (size_t i = 0; i < size; i++)
+		buf[i] = rand() & 0xff;
+}
+
+TraversalClient::TraversalClient(ENetHost* netHost, const std::string& server)
+	: m_NetHost(netHost)
+	, m_Server(server)
+	, m_Client(NULL)
+	, m_FailureReason(0)
+	, m_ConnectRequestId(0)
+	, m_PendingConnect(false)
+	, m_PingTime(0)
+{
+	netHost->intercept = TraversalClient::InterceptCallback;
+
+	Reset();
+
+	ReconnectToServer();
+}
+
+TraversalClient::~TraversalClient()
+{
+}
+
+void TraversalClient::ReconnectToServer()
+{
+	m_Server = "vps.qoid.us"; // XXX
+	if (enet_address_set_host(&m_ServerAddress, m_Server.c_str()))
+	{
+		OnFailure(BadHost);
+		return;
+	}
+	m_ServerAddress.port = 6262;
```

Comment:
Sounds like we might be moving this to another vpn, so maybe something found from an ini instead?